pallets/ajuna-awesome-avatars/src/lib.rs

```diff
@@ -1101,6 +1126,65 @@ pub mod pallet {
 				},
 			}
 		}
+
+		#[pallet::call_index(22)]
+		#[pallet::weight({1000})]
+		pub fn add_affiliation(origin: OriginFor<T>, affiliate_id: u64) -> DispatchResult {
+			let account = ensure_signed(origin)?;
+
+			if let Some(affiliator) = T::AffiliateHandler::get_account_for_id(affiliate_id) {
+				T::AffiliateHandler::try_add_affiliate_to(&affiliator, &account)
+			} else {
+				Err(Error::<T>::AffiliatorNotFound.into())
+			}
+		}
+
+		#[pallet::call_index(23)]
+		#[pallet::weight({1000})]
+		pub fn enable_affiliator(
+			origin: OriginFor<T>,
+			target: AffiliatorTarget<T::AccountId>,
+		) -> DispatchResult {
+			let account = ensure_signed(origin)?;
+
+			match target {
+				AffiliatorTarget::OneselfFree => {
+					// Check criteria
+					// T::AffiliateHandler::try_mark_account_as_affiliatable(&account)?;
+				},
+				AffiliatorTarget::OneselfPaying => {
+					// Substract amout for paying if account not affiliator
+					let GlobalConfig { affiliate_config, .. } = GlobalConfigs::<T>::get();
+					T::Currency::transfer(
+						&account,
+						&Self::treasury_account_id(),
+						affiliate_config.affiliator_enable_fee,
+						AllowDeath,
+					)?;
+					T::AffiliateHandler::try_mark_account_as_affiliatable(&account)?;
+				},
+				AffiliatorTarget::OtherPaying(other) => {
+					// Substract amout for paying if other not affiliator
+					let GlobalConfig { affiliate_config, .. } = GlobalConfigs::<T>::get();
+					T::Currency::transfer(
+						&account,
+						&Self::treasury_account_id(),
+						affiliate_config.affiliator_enable_fee,
+						AllowDeath,
+					)?;
+					T::AffiliateHandler::try_mark_account_as_affiliatable(&other)?;
+				},
+			}
+
+			Ok(())
+		}
+
+		#[pallet::call_index(24)]
+		#[pallet::weight({1000})]
+		pub fn remove_affiliation(origin: OriginFor<T>) -> DispatchResult {
+			let account = ensure_signed(origin)?;
```

Comment:
removing affiliation let it be an organizer call and be callable for any address